src/Components/Articles.jsx

```diff
@@ -5,14 +5,34 @@ import ArticleContent from './ArticleContent'
 class Articles extends Component {
   state = {
     articles: [],
-    singleArticle: null
+    singleArticle: null,
   };
 
-  componentDidMount = async () => {
-    let response = await axios.get(`/articles`);
-    this.setState({ articles: response.data.articles });
+  componentDidUpdate = (prevProps) => {
+    if (this.props.history.location.pathname !== prevProps.location.pathname) {
+      this.setState({ singleArticle: null })
+      this.getArticles()
+    }
+  }
+
+  componentDidMount = () => {
+    this.getArticles()
   };
 
+  getArticles = async () => {
+    let response
+    if (this.props.history.location.pathname === "/") {
+      response = await axios.get(`/articles`);
+    } else {
+      response = await axios.get(`/articles`,
+        {
+          params: this.props.match.params
+        }
```

Comment:
I would try this request while running the backend to make sure it is working, if you haven't done that. To make sure you don't get the params keyword twice in the request